packages/matchbox/src/components/Banner/Banner.js

```diff
@@ -1,144 +1,147 @@
+
+  const titleMarkup = title ? (
+    <Box pt="200" mb="200">
+      <Text fontSize="500" lineHeight="500" as="h5">
+        {title}
+      </Text>
+    </Box>
+  ) : null;
 
-class Banner extends Component {
-  static displayName = 'Banner';
-
-  static propTypes = {
-    /**
-     * The type of banner. 'default' | 'success' | 'warning' | 'danger' | 'info'
-     */
-    status: PropTypes.oneOf(['default', 'success', 'warning', 'danger', 'info']),
-
-    /**
-     * The banner's title
-     */
-    title: PropTypes.string,
-
-    /**
-     * Callback when dismiss button is clicked. Button hidden without callback.
-     */
-    onDismiss: PropTypes.func,
-
-    /**
-     * Action that build a button. Most button props will work in here.
-     * e.g. { content: 'button label', onClick: callback() }
-     */
-    action: PropTypes.shape({ content: PropTypes.string.isRequired }),
-
-    /**
-     * List of actions that build buttons. Most button props will work in here.
-     * Overrides `action`
-     */
-    actions: PropTypes.arrayOf(PropTypes.shape({
-      content: PropTypes.string.isRequired
-    })),
-
-    /**
-     * Banner Content
-     */
-    children: PropTypes.node
-  };
-
-  static defaultProps = {
-    status: 'default'
-  };
-
-  render() {
-    const {
-      children,
-      title,
-      status,
-      action,
-      actions,
-      onDismiss,
-      ...rest
-    } = this.props;
-
-    const titleMarkup = title
-      ? <h5 className={styles.Title}>{title}</h5>
-      : null;
-
-    let actionMarkup = action
-      ? <div className={styles.Actions}>{buttonFrom(action, actionOverrides)}</div>
-      : null;
+  const actionMarkup = React.useMemo(() => {
+    let result = action ? <div>{buttonFrom(action)}</div> : null;
 
     if (actions) {
-      actionMarkup = (
-        <div className={styles.Actions}>
-          {actions.map((action, i) => buttonFrom(action, actionOverrides, i))}
-        </div>
-      );
+      result = <Inline>{actions.map((action, i) => buttonFrom(action, {}, i))}</Inline>;
     }
 
-    const dismissMarkup = onDismiss
-      ? <a className={styles.Dismiss} onClick={onDismiss}><Close size={24} className={styles.DismissIcon} /></a>
-      : null;
-
-    const bannerStyles = classnames(
-      styles.Banner,
-      styles[`${status}`]
-    );
-
-    return (
-      <div className={bannerStyles} {...rest}>
-        <IconSection status={status} />
-        <div className={styles.Content}>
-          {dismissMarkup}
-          {titleMarkup}
-          <div className={styles.Children}>{children}</div>
-          {actionMarkup}
-        </div>
-      </div>
-    );
-  }
+    return result;
+  }, [action, actions]);
+
+  const dismissMarkup = onDismiss ? (
+    <Box flex="0">
+      <StyledDismiss as="button" onClick={onDismiss} status={status} color="gray.800">
+        <ScreenReaderOnly>Dismiss</ScreenReaderOnly>
+        <Close size={24} />
+      </StyledDismiss>
+    </Box>
+  ) : null;
```

Comment:
In the future, I wonder if we could use components for this sort of thing instead of variables? This is really not a substantive critique. In general FE devs use variables and functions within the scope of a component for this sort of thing, when I feel like just adding additional (but not exported) components might make more sense and make things more "React-y"? Worth a convo, but not worth changing anything at this point.

Reply:
Yup agree 100%